src/components/select-field/BaseSelectField.js

```diff
@@ -256,9 +257,17 @@ class BaseSelectField extends React.Component<Props, State> {
                 break;
             case 'Enter':
             case ' ':
-                // When the search input is active, we want to allow space key presses in the search string
-                if (key === ' ' && shouldShowSearchInput) {
-                    break;
+                if (shouldShowSearchInput) {
+                    // Allow space key presses in the search string when search field is active
+                    if (key === ' ') {
+                        break;
+                    }
+
+                    // Enter presses should be ignored when no item is active
+                    if (key === 'Enter' && activeItemIndex === -1) {
+                        stopDefaultEvent(event);
```

Comment:
This change is needed because we do not want the 'Enter' keypress to submit the form if there is no active item selected. The bug that was happening was that when the search field was empty (or when no results were shown) and the user pressed the Enter key, the page would reload. We want to have Enter keypresses ignored when Enter is pressed and no item is highlighted.